package/MDAnalysis/analysis/rms.py

```diff
@@ -546,7 +546,7 @@ def __init__(self, atomgroup, reference=None, select='all',
         # check weights type
         if iterable(self.weights) and (np.array(weights).dtype
                                 not in (np.dtype('float64'),np.dtype('int64'))):
-            raise TypeError("weight should only be be 'mass', None or 1D float array."
+            raise ValueError("weight should only be be 'mass', None or 1D float array."
```

Comment:
This should still be a `TypeError` because it is raised when the `dtype` is incorrect.

Reply:
Almost missed it, Thank you for pointing that out.